File: solutions/347-Top-K-Frequent-Elements.js
var topKFrequent = function(nums,k){
    let length = nums.length
    let map = {}
    let ret = []
    for(let i=0;i<length;i++)
        if(map[nums[i]]) map[nums[i]] ++
        else map[nums[i]] = 1
    let keys = []
    for(let key in map) keys.push(key)
    keys.sort((key1,key2)=>map[key2]-map[key1])
    for(let i=0;i<k;i++) ret.push(+keys[i])
    return ret
}

File: solutions/347-top-k-frequent-elements.cpp
#include <iostream>
#include <queue>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
 public:
  vector<int> topKFrequent(vector<int>& nums, int k) {
    unordered_map<int, int> map;
    auto comp = [&](int a, int b) { return map[a] < map[b]; };
    priority_queue<int, vector<int>, decltype(comp)> p_queue(comp);
    vector<int> result;
    for (int& num : nums) {
      map[num]++;
    }
    for (auto& [key, value] : map) {
      p_queue.push(key);
    }
    for (int i = 0; i < k; i++) {
      result.push_back(p_queue.top());
      p_queue.pop();
    }
    return result;
  }
};

int main() {
  Solution s;
  vector<int> v{1, 1, 1, 2, 2, 3};
  auto r = s.topKFrequent(v, 2);
  for (int& n : r) {
    cout << n << endl;
  }
  return 0;
}